### ISSUE TITLE
'compose service up' does not ensure service is created before trying to start it

### ISSUE THREAD

**Issue by perryn** [aws/amazon-ecs-cli#79]
I just ran 'compose service up' which failed with the error

"Please use 'create' command to create the service 'my-service-name' first"

Usually it works, but it appears that it is possible under some circumstances for 'up' to  try to 'start' when 'create' has failed (or at least before it is complete)

**Comment by andeero** [comment]
I'm seeing this myself. But the output suggests the service is created. Happens sporadically.

`+ ecs-cli compose --file docker-compose.yml --project-name oracle-xe service up

time="2016-10-26T08:41:31Z" level=info msg="Using ECS task definition" TaskDefinition="oracle-xe:6" 

time="2016-10-26T08:41:31Z" level=info msg="Created an ECS service" service=jenkins-someproject-integration-test-48-oracle-xe taskDefinition="oracle-xe:6" 

time="2016-10-26T08:41:31Z" level=fatal msg="Please use 'create' command to create the service 'jenkins-someproject-integration-test-48-oracle-xe' first" `

**Comment by bastianb** [comment]
+1

We are also seeing the exact same issue:
`ecs-cli version 0.4.4 (7e1376e)`

Randomly throwing this error but does create the service - without any task then fails
Next run it works perfectly

```
$ ecs-cli compose --file docker/docker-compose.testenv.yml service up
time="2017-01-20T10:57:41Z" level=info msg="Using ECS task definition" TaskDefinition="ecscompose-testenv-ready-newconsul-516:1" 
time="2017-01-20T10:57:41Z" level=info msg="Created an ECS service" service=ecscompose-service-testenv-ready-newconsul-516 taskDefinition="ecscompose-testenv-ready-newconsul-516:1" 
time="2017-01-20T10:57:41Z" level=fatal msg="Please use 'create' command to create the service 'ecscompose-service-testenv-ready-newconsul-516' first" 
```

**Comment by Lowess** [comment]
Is there any progress on this issue ? I see that a PR is waiting for reviewers in order to be merge. Any plan to address this soon ?

We are heavily relying on ecs-cli in our continuous deployment pipelines and we are facing this issue quite often.

Thanks !

**Comment by SoManyHs** [comment]
Hi @Lowess,
We are currently in the process of reviewing the PR and we'll let you know soon!

**Comment by Lowess** [comment]
Any plan for a release soon ? This bug fix will be really helpful !

Thanks !

**Comment by SoManyHs** [comment]
@Lowess This is fixed in v1.7.0!
